src/app/business-details/business-details.component.ts

```diff
@@ -1,15 +1,32 @@
 import { Component, OnInit } from '@angular/core';
+import { Business } from '../../backend/model';
+import { ActivatedRoute } from '@angular/router';
+import { BusinessesService } from '../../backend/services';
 
 @Component({
-  selector: 'app-business-details',
-  templateUrl: './business-details.component.html',
-  styleUrls: ['./business-details.component.scss']
+    selector: 'app-business-details',
+    templateUrl: './business-details.component.html',
+    styleUrls: ['./business-details.component.scss']
 })
 export class BusinessDetailsComponent implements OnInit {
+    business: Business;
 
-  constructor() { }
+    constructor(
+        private route: ActivatedRoute,
+        private businessesService: BusinessesService
+    ) {
+    }
 
-  ngOnInit() {
-  }
+    ngOnInit() {
+        this.route.params.subscribe(params => {
+            if (params.id) {
+                this.businessesService.get(+params.id).subscribe(business => {
```

Comment:
`this.businessesService.get(+params.id).subscribe(business => this.business = business);`